diff -Naur a/dev_appserver.py b/dev_appserver.py
--- a/dev_appserver.py	2010-04-19 20:02:11.000000000 +0200
+++ b/dev_appserver.py	2010-04-22 10:26:30.000000000 +0200
@@ -43,6 +43,8 @@
   os.path.join(DIR_PATH, 'lib', 'ipaddr'),
   os.path.join(DIR_PATH, 'lib', 'webob'),
   os.path.join(DIR_PATH, 'lib', 'yaml', 'lib'),
+  os.path.join(DIR_PATH, '..', '..', 'src', 'redis'),
+  os.path.join(DIR_PATH, '..', '..', 'src', 'typhoonae', 'redis'),
 ]
 
 SCRIPT_EXCEPTIONS = {
diff -Naur a/google/appengine/tools/dev_appserver.py b/google/appengine/tools/dev_appserver.py
--- a/google/appengine/tools/dev_appserver.py	2010-04-19 20:02:24.000000000 +0200
+++ b/google/appengine/tools/dev_appserver.py	2010-04-22 10:29:41.000000000 +0200
@@ -2297,7 +2297,7 @@
     __builtin__.open = FakeFile
     types.FileType = FakeFile
 
-    __builtin__.buffer = NotImplementedFakeClass
+    #__builtin__.buffer = NotImplementedFakeClass
 
     logging.debug('Executing CGI with env:\n%s', pprint.pformat(env))
     try:
@@ -3519,6 +3519,7 @@
   datastore_path = config['datastore_path']
   clear_datastore = config['clear_datastore']
   use_sqlite = config.get('use_sqlite', False)
+  use_redis = config.get('use_redis', False)
   require_indexes = config.get('require_indexes', False)
   smtp_host = config.get('smtp_host', None)
   smtp_port = config.get('smtp_port', 25)
@@ -3546,6 +3547,19 @@
     datastore = datastore_sqlite_stub.DatastoreSqliteStub(
         app_id, datastore_path, require_indexes=require_indexes,
         trusted=trusted)
+  if use_redis:
+    from google.appengine.datastore import datastore_index    
+    import datastore_redis_stub
+    try:
+      index_yaml = open(os.path.join(root_path, 'index.yaml'), 'r')
+      indexes = datastore_index.IndexDefinitionsToProtos(
+        app_id, datastore_index.ParseIndexDefinitions(index_yaml).indexes)
+    except IOError:
+      indexes = []
+    except TypeError:
+      indexes = []
+    datastore = datastore_redis_stub.DatastoreRedisStub(
+        app_id, indexes)
   else:
     datastore = datastore_file_stub.DatastoreFileStub(
         app_id, datastore_path, require_indexes=require_indexes,
diff -Naur a/google/appengine/tools/dev_appserver_main.py b/google/appengine/tools/dev_appserver_main.py
--- a/google/appengine/tools/dev_appserver_main.py	2010-04-19 20:02:26.000000000 +0200
+++ b/google/appengine/tools/dev_appserver_main.py	2010-04-22 10:49:00.000000000 +0200
@@ -34,6 +34,8 @@
                              (Default %(datastore_path)s)
   --use_sqlite               Use the new, SQLite based datastore stub.
                              (Default false)
+  --use_redis                Use TyphoonAE's Redis Datastore stub.
+                             (Default false)
   --history_path=PATH        Path to use for storing Datastore history.
                              (Default %(history_path)s)
   --require_indexes          Disallows queries that require composite indexes
@@ -97,6 +99,7 @@
 ARG_BLOBSTORE_PATH = 'blobstore_path'
 ARG_DATASTORE_PATH = 'datastore_path'
 ARG_USE_SQLITE = 'use_sqlite'
+ARG_USE_REDIS = 'use_redis'
 ARG_DEBUG_IMPORTS = 'debug_imports'
 ARG_ENABLE_SENDMAIL = 'enable_sendmail'
 ARG_SHOW_MAIL_BODY = 'show_mail_body'
@@ -130,6 +133,7 @@
   ARG_DATASTORE_PATH: os.path.join(tempfile.gettempdir(),
                                    'dev_appserver.datastore'),
   ARG_USE_SQLITE: False,
+  ARG_USE_REDIS: False,
   ARG_HISTORY_PATH: os.path.join(tempfile.gettempdir(),
                                  'dev_appserver.datastore.history'),
   ARG_LOGIN_URL: '/_ah/login',
@@ -194,6 +198,7 @@
         'blobstore_path=',
         'datastore_path=',
         'use_sqlite',
+        'use_redis',
         'debug',
         'debug_imports',
         'enable_sendmail',
@@ -242,6 +247,9 @@
     if option == '--use_sqlite':
       option_dict[ARG_USE_SQLITE] = True
 
+    if option in '--use_redis':
+      option_dict[ARG_USE_REDIS] = True
+
     if option == '--history_path':
       option_dict[ARG_HISTORY_PATH] = os.path.abspath(value)
 
