diff -Naur a/dev_appserver.py b/dev_appserver.py
--- a/dev_appserver.py	2010-03-23 00:34:38.000000000 +0100
+++ b/dev_appserver.py	2010-04-04 02:50:37.000000000 +0200
@@ -43,6 +43,8 @@
   os.path.join(DIR_PATH, 'lib', 'ipaddr'),
   os.path.join(DIR_PATH, 'lib', 'webob'),
   os.path.join(DIR_PATH, 'lib', 'yaml', 'lib'),
+  os.path.join(DIR_PATH, '..', '..', 'src', 'redis'),
+  os.path.join(DIR_PATH, '..', '..', 'src', 'typhoonae', 'redis'),
 ]
 
 SCRIPT_EXCEPTIONS = {
diff -Naur a/google/appengine/tools/dev_appserver.py b/google/appengine/tools/dev_appserver.py
--- a/google/appengine/tools/dev_appserver.py	2010-03-23 00:34:48.000000000 +0100
+++ b/google/appengine/tools/dev_appserver.py	2010-04-04 10:57:50.000000000 +0200
@@ -2290,7 +2290,7 @@
     __builtin__.open = FakeFile
     types.FileType = FakeFile
 
-    __builtin__.buffer = NotImplementedFakeClass
+    #__builtin__.buffer = NotImplementedFakeClass
 
     logging.debug('Executing CGI with env:\n%s', pprint.pformat(env))
     try:
@@ -3518,6 +3518,7 @@
   enable_sendmail = config.get('enable_sendmail', False)
   show_mail_body = config.get('show_mail_body', False)
   remove = config.get('remove', os.remove)
+  use_redis = config.get('use_redis', False)
   trusted = config.get('trusted', False)
 
   os.environ['APPLICATION_ID'] = app_id
@@ -3533,9 +3534,25 @@
 
   apiproxy_stub_map.apiproxy = apiproxy_stub_map.APIProxyStubMap()
 
-  datastore = datastore_file_stub.DatastoreFileStub(
-      app_id, datastore_path, require_indexes=require_indexes,
-      trusted=trusted)
+  if use_redis:
+    from google.appengine.datastore import datastore_index    
+    import datastore_redis_stub
+    try:
+      index_yaml = open(os.path.join(root_path, 'index.yaml'), 'r')
+      indexes = datastore_index.IndexDefinitionsToProtos(
+          'test',
+          datastore_index.ParseIndexDefinitions(index_yaml).indexes)
+    except IOError:
+      indexes = []
+    except TypeError:
+      indexes = []
+    datastore = datastore_redis_stub.DatastoreRedisStub(
+        app_id, indexes)
+  else:
+    datastore = datastore_file_stub.DatastoreFileStub(
+        app_id, datastore_path, require_indexes=require_indexes,
+        trusted=trusted)
+    
   apiproxy_stub_map.apiproxy.RegisterStub('datastore_v3', datastore)
 
   fixed_login_url = '%s?%s=%%s' % (login_url,
diff -Naur a/google/appengine/tools/dev_appserver_main.py b/google/appengine/tools/dev_appserver_main.py
--- a/google/appengine/tools/dev_appserver_main.py	2010-03-23 00:34:49.000000000 +0100
+++ b/google/appengine/tools/dev_appserver_main.py	2010-04-04 02:50:37.000000000 +0200
@@ -59,6 +59,7 @@
                              skipped_files (default False)
   --disable_static_caching   Never allow the browser to cache static files.
                              (Default enable if expiration set in app.yaml)
+  --use_redis                Use Redis Datastore backend.
 """
 
 
@@ -110,6 +111,7 @@
 ARG_STATIC_CACHING = 'static_caching'
 ARG_TEMPLATE_DIR = 'template_dir'
 ARG_TRUSTED = 'trusted'
+ARG_USE_REDIS = 'use_redis'
 
 SDK_PATH = os.path.dirname(
              os.path.dirname(
@@ -204,6 +206,7 @@
         'smtp_user=',
         'template_dir=',
         'trusted',
+        'use_redis',
       ])
   except getopt.GetoptError, e:
     print >>sys.stderr, 'Error: %s' % e
@@ -291,6 +294,9 @@
     if option == '--trusted':
       option_dict[ARG_TRUSTED] = True
 
+    if option in '--use_redis':
+      option_dict[ARG_USE_REDIS] = True
+
   return args, option_dict
 
 
